src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -411,15 +403,42 @@ private static bool IsInaccessibleImplementableAccessor(IMethodSymbol? accessor,
             Func<INamedTypeSymbol, ISymbol, ImmutableArray<ISymbol>> interfaceMemberGetter,
             CancellationToken cancellationToken)
         {
-            var q = from m in interfaceMemberGetter(interfaceType, classOrStructType)
-                    where m.Kind != SymbolKind.NamedType
-                    where m.Kind != SymbolKind.Method || ((IMethodSymbol)m).MethodKind is MethodKind.Ordinary or MethodKind.UserDefinedOperator or MethodKind.Conversion
-                    where m.Kind != SymbolKind.Property || ((IPropertySymbol)m).IsIndexer || ((IPropertySymbol)m).CanBeReferencedByName
-                    where m.Kind != SymbolKind.Event || ((IEventSymbol)m).CanBeReferencedByName
-                    where !isImplemented(classOrStructType, m, isValidImplementation, cancellationToken)
-                    select m;
-
-            return q.ToImmutableArray();
+            using var _ = ArrayBuilder<ISymbol>.GetInstance(out var results);
+
+            foreach (var member in interfaceMemberGetter(interfaceType, classOrStructType))
+            {
+                switch (member)
+                {
+                    case IPropertySymbol property:
+                        if (property.IsIndexer || property.CanBeReferencedByName)
+                            AddIfNotImplemented(property);
+
+                        break;
+
+                    case IEventSymbol ev:
+                        if (ev.CanBeReferencedByName)
+                            AddIfNotImplemented(ev);
+
+                        break;
+
+                    case IMethodSymbol method:
+                        if (method is { MethodKind: MethodKind.UserDefinedOperator or MethodKind.Conversion } ||
+                            method is { MethodKind: MethodKind.Ordinary, CanBeReferencedByName: true })
```

Comment:
this line is the fix.  ordinary methods have to be referenceable by name if we want to be able to override them.